In [1]:
import numpy as np 
import pandas as pd

In [2]:
data = pd.read_csv(r"C:\Users\ladm\Downloads\data.csv")

In [3]:
data.head()

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  2006
0         Demand   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
1           OECD  46.9  47.8  47.9  47.9  47.9  48.6  49.4  49.7  49.3
2  North America  23.1  23.8  24.1  24.0  24.1  24.5  25.4  25.5  25.3
3         Europe  15.4  15.3  15.2  15.4  15.3  15.4  15.5  15.6  15.6
4        Pacific   8.4   8.7   8.6   8.5   8.5   8.6   8.5   8.6   8.4

In [4]:
data

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  \
0                     Demand   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1                       OECD  46.9  47.8  47.9  47.9  47.9  48.6  49.4  49.7   
2              North America  23.1  23.8  24.1  24.0  24.1  24.5  25.4  25.5   
3                     Europe  15.4  15.3  15.2  15.4  15.3  15.4  15.5  15.6   
4                    Pacific   8.4   8.7   8.6   8.5   8.5   8.6   8.5   8.6   
5                   Non-OECD  27.3  28.0  28.6  29.2  29.8  30.7  33.0  34.1   
6                      Total  74.2  75.8  76.5  77.1  77.7  79.3  82.3  83.7   
7                     Supply   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
8                       OECD  21.9  21.5  21.9  21.8  21.9  21.6  21.2  20.3   
9                 OPEC total  31.0  29.6  31.0  30.5  28.9  30.8  33.1  34.2   
10               Formar USSR   7.3   7.5   7.9   8.6   9.4  10.3  11.2  11.6   
11            Other non-OECD  15.7  16.0  16.2  16.4  16.9  17.1  17.7  18.2   
12                     Total  75.9  74.5  77.1  77.3  77.0  79.8  83.2  84.4   
13  Brent crude import price  12.7  17.9  28.4  24.5  25.0  28.0  38.2  54.4   

    2006  
0    NaN  
1   49.3  
2   25.3  
3   15.6  
4    8.4  
5   35.4  
6   84.7  
7    NaN  
8   20.0  
9   34.3  
10  12.2  
11  18.7  
12  85.3  
13  65.1

## a. What was the percentage increase in the price of oil between 1998 and 2006?

In [5]:
# Percentage increase in price = (Change in price/original price) * 100
percent= (data['2006'].iloc[13] - 
                  data['1998'].iloc[13])/data['1998'].iloc[13] * 100
print(percent)

412.59842519685037


## The percentage increase in the price of oil between 1998 and 2006 is 412.598425% which roundoffs to 413% .

## b.If the total oil supply had grown at the same rate as the former USSR oil supply between 2000 and 2006 how much would this have exceeded the demand in 2006?

In [6]:
data

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  \
0                     Demand   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1                       OECD  46.9  47.8  47.9  47.9  47.9  48.6  49.4  49.7   
2              North America  23.1  23.8  24.1  24.0  24.1  24.5  25.4  25.5   
3                     Europe  15.4  15.3  15.2  15.4  15.3  15.4  15.5  15.6   
4                    Pacific   8.4   8.7   8.6   8.5   8.5   8.6   8.5   8.6   
5                   Non-OECD  27.3  28.0  28.6  29.2  29.8  30.7  33.0  34.1   
6                      Total  74.2  75.8  76.5  77.1  77.7  79.3  82.3  83.7   
7                     Supply   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
8                       OECD  21.9  21.5  21.9  21.8  21.9  21.6  21.2  20.3   
9                 OPEC total  31.0  29.6  31.0  30.5  28.9  30.8  33.1  34.2   
10               Formar USSR   7.3   7.5   7.9   8.6   9.4  10.3  11.2  11.6   
11            Other non-OECD  15.7  16.0  16.2  16.4  16.9  17.1  17.7  18.2   
12                     Total  75.9  74.5  77.1  77.3  77.0  79.8  83.2  84.4   
13  Brent crude import price  12.7  17.9  28.4  24.5  25.0  28.0  38.2  54.4   

    2006  
0    NaN  
1   49.3  
2   25.3  
3   15.6  
4    8.4  
5   35.4  
6   84.7  
7    NaN  
8   20.0  
9   34.3  
10  12.2  
11  18.7  
12  85.3  
13  65.1

In [7]:
#change_in_supply = (supply in 2006/supply in 2000)*100
change_in_supply= (data['2006'].iloc[10]/data['2000'].iloc[10])* 100
print(change_in_supply)

154.43037974683543


In [8]:
#supply_in_2006 = total supply in 2000 x change in supply for USSR 2000-2006
supply_in_2006 = (data['2000'].iloc[12])*(change_in_supply/100)
print(supply_in_2006)

119.0658227848101


In [9]:
#difference = supply_in_2006 - demand in 2006
difference = supply_in_2006 - data['2006'].iloc[6]
print(difference)

34.36582278481009


## It would have exceeded by 34.3

## c.In which year was the largest combined change in supply from the previous year for OPEC, the former USSR and other Non-OECD countries?

In [10]:
data

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  \
0                     Demand   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1                       OECD  46.9  47.8  47.9  47.9  47.9  48.6  49.4  49.7   
2              North America  23.1  23.8  24.1  24.0  24.1  24.5  25.4  25.5   
3                     Europe  15.4  15.3  15.2  15.4  15.3  15.4  15.5  15.6   
4                    Pacific   8.4   8.7   8.6   8.5   8.5   8.6   8.5   8.6   
5                   Non-OECD  27.3  28.0  28.6  29.2  29.8  30.7  33.0  34.1   
6                      Total  74.2  75.8  76.5  77.1  77.7  79.3  82.3  83.7   
7                     Supply   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
8                       OECD  21.9  21.5  21.9  21.8  21.9  21.6  21.2  20.3   
9                 OPEC total  31.0  29.6  31.0  30.5  28.9  30.8  33.1  34.2   
10               Formar USSR   7.3   7.5   7.9   8.6   9.4  10.3  11.2  11.6   
11            Other non-OECD  15.7  16.0  16.2  16.4  16.9  17.1  17.7  18.2   
12                     Total  75.9  74.5  77.1  77.3  77.0  79.8  83.2  84.4   
13  Brent crude import price  12.7  17.9  28.4  24.5  25.0  28.0  38.2  54.4   

    2006  
0    NaN  
1   49.3  
2   25.3  
3   15.6  
4    8.4  
5   35.4  
6   84.7  
7    NaN  
8   20.0  
9   34.3  
10  12.2  
11  18.7  
12  85.3  
13  65.1

In [11]:
change_data = data.iloc[[9,10,11]]
change_data

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  2006
9       OPEC total  31.0  29.6  31.0  30.5  28.9  30.8  33.1  34.2  34.3
10     Formar USSR   7.3   7.5   7.9   8.6   9.4  10.3  11.2  11.6  12.2
11  Other non-OECD  15.7  16.0  16.2  16.4  16.9  17.1  17.7  18.2  18.7

In [12]:
change_data.loc['Total'] = (change_data.sum(axis=0))
change_data

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value)


OIL MARKETS  1998  1999  2000  2001  2002  \
9                               OPEC total  31.0  29.6  31.0  30.5  28.9   
10                             Formar USSR   7.3   7.5   7.9   8.6   9.4   
11                          Other non-OECD  15.7  16.0  16.2  16.4  16.9   
Total  OPEC totalFormar USSROther non-OECD  54.0  53.1  55.1  55.5  55.2   

       2003  2004  2005  2006  
9      30.8  33.1  34.2  34.3  
10     10.3  11.2  11.6  12.2  
11     17.1  17.7  18.2  18.7  
Total  58.2  62.0  64.0  65.2

In [13]:
change_data=change_data.replace(to_replace ="OPEC totalFormar USSROther non-OECD",
                 value ="-")
change_data

OIL MARKETS  1998  1999  2000  2001  2002  2003  2004  2005  2006
9          OPEC total  31.0  29.6  31.0  30.5  28.9  30.8  33.1  34.2  34.3
10        Formar USSR   7.3   7.5   7.9   8.6   9.4  10.3  11.2  11.6  12.2
11     Other non-OECD  15.7  16.0  16.2  16.4  16.9  17.1  17.7  18.2  18.7
Total               -  54.0  53.1  55.1  55.5  55.2  58.2  62.0  64.0  65.2

In [14]:
change_data.loc['Combined Change'] = (change_data).sum(axis=0)
change_data['1998'].iloc[4] = "-"
change_data['1999'].iloc[4]=change_data['1999'].iloc[3]-change_data['1998'].iloc[3]
change_data['2000'].iloc[4]=change_data['2000'].iloc[3]-change_data['1999'].iloc[3]
change_data['2001'].iloc[4]=change_data['2001'].iloc[3]-change_data['2000'].iloc[3]
change_data['2002'].iloc[4]=change_data['2002'].iloc[3]-change_data['2001'].iloc[3]
change_data['2003'].iloc[4]=change_data['2003'].iloc[3]-change_data['2002'].iloc[3]
change_data['2004'].iloc[4]=change_data['2004'].iloc[3]-change_data['2003'].iloc[3]
change_data['2005'].iloc[4]=change_data['2005'].iloc[3]-change_data['2004'].iloc[3]
change_data['2006'].iloc[4]=change_data['2006'].iloc[3]-change_data['2005'].iloc[3]
change_data

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value)


OIL MARKETS  1998  1999  2000  2001  \
9                                          OPEC total    31  29.6  31.0  30.5   
10                                        Formar USSR   7.3   7.5   7.9   8.6   
11                                     Other non-OECD  15.7  16.0  16.2  16.4   
Total                                               -    54  53.1  55.1  55.5   
Combined Change  OPEC totalFormar USSROther non-OECD-     -  -0.9   2.0   0.4   

                 2002  2003  2004  2005  2006  
9                28.9  30.8  33.1  34.2  34.3  
10                9.4  10.3  11.2  11.6  12.2  
11               16.9  17.1  17.7  18.2  18.7  
Total            55.2  58.2  62.0  64.0  65.2  
Combined Change  -0.3   3.0   3.8   2.0   1.2

## Since the highest combined change is 3.8 in year 2004. So the year is 2004